src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,94 +55,1315 @@
+        
+        for (String possibleRecipient : possibleReceivers) {
+            String possibleRecipientName = bundle.getFullNameFromRoster(possibleRecipient);
+            String possibleRecipientTeam = bundle.getTeamNameFromRoster(possibleRecipient);
+            
+            String textToDisplay = questionDetails.getNoResponseTextInHtml(giverIdentifier, possibleRecipient, bundle, question);
+            
+            if (questionDetails.shouldShowNoResponseText(giverIdentifier, possibleRecipient, question)) {
+                InstructorFeedbackResultsModerationButton moderationButton = buildModerationButtonForGiver(
+                                                                         question, giverIdentifier, 
+                                                                         "btn btn-default btn-xs", "Moderate Response");
+                InstructorFeedbackResultsResponseRow missingResponse = new InstructorFeedbackResultsResponseRow(giverName, giverTeam, 
+                                                                                                possibleRecipientName, possibleRecipientTeam, 
+                                                                                                textToDisplay, moderationButton, true);
+                
+                missingResponse.setRowAttributes(new ElementTag("class", "pending_response_row"));
+                configureResponseRow(question, giverIdentifier, possibleRecipient, missingResponse);
+                missingResponses.add(missingResponse);
             }
-        } else if (bundle.feedbackSession.resultsVisibleFromTime.equals(Const.TIME_REPRESENTS_LATER)) {
-            return "I want to manually publish the results.";
-        } else if (bundle.feedbackSession.resultsVisibleFromTime.equals(Const.TIME_REPRESENTS_NEVER)) {
-            return "Never";
+        }
+        
+        return missingResponses;
+    }
+    
+    /**
+     * Construct missing response rows between the recipient identified by {@code recipientIdentifier} and 
+     * {@code possibleGivers}. The response rows are configured using 
+     * {@code configureResponseRowForViewType(viewType)}. 
+     *  
+     * @see configureResponseRowForViewType
+     */
+    private List<InstructorFeedbackResultsResponseRow> buildMissingResponseRowsBetweenRecipientAndPossibleGivers(
+                                    FeedbackQuestionAttributes question, 
+                                    List<String> possibleGivers, String recipientIdentifier,
+                                    String recipientName, String recipientTeam) {
+        List<InstructorFeedbackResultsResponseRow> missingResponses = new ArrayList<InstructorFeedbackResultsResponseRow>();
+        FeedbackQuestionDetails questionDetails = question.getQuestionDetails();
+        
+        for (String possibleGiver : possibleGivers) {
+            String possibleGiverName = bundle.getFullNameFromRoster(possibleGiv
```

Comment:
Are both `isAllowedToModerate` and `isDisabled` necessary?
One seems to just be the inverse of the other (the constructor for the button is only used here).